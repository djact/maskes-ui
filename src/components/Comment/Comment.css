.ui.comments .comment .avatar img{
    border-radius: 2rem;
}

.ui.comments .comment .actions .edit-btn:hover{
    cursor: pointer;
    display: inline-block;
    margin: 0 .75em 0 0;
    color: teal;
}

.ui.comments .comment .actions .delete-btn{
    color: rgb(255, 81, 81);
}
.ui.comments .comment .actions .delete-btn:hover{
    color: red;
    font-size: 14px;
    font-weight: bold;
}

.post-reply-btn {
    background-color: transparent;
    color: #7cb147;
    border: none;
    cursor: pointer;
    display: inline;
    margin: 2px;
    padding: 0;
    font-size: 18px;
}

.post-reply-btn:hover{
    color: green;
    font-weight: bold;
}

.delete-reply-btn{
    background-color: transparent;
    color: salmon;
    border: none;
    cursor: pointer;
    display: inline;
    margin: 2px;
    padding: 0;
    font-size: 18px;
}
.delete-reply-btn:hover{
    color: red;
    font-weight: bold;
}
