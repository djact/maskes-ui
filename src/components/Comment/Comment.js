import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { Comment as CommentSUI, Form } from 'semantic-ui-react';
import { Button as BSButton } from 'react-bootstrap';
import { FiChevronDown } from 'react-icons/fi';
import { BsPencil, BsXSquare } from 'react-icons/bs';
import Aux from '../../hoc/Aux/Aux';
import './Comment.css';
import Replies from '../../containers/Replies/Replies';

const Comment = (props) => {
	const {
		comment,
		update,
		userId,
		moment,
		setDeleteCommentId,
		setShowDeleteModal
	} = props;

	const history = useHistory();

	//Edit Comment Form
	const [onEdit, setOnEdit] = useState(false);
	const toggleEditor = () => {
		setOnEdit(!onEdit);
	};

	const [commentContent, setCommentContent] = useState(comment.comment_content);
	const onChangeComment = (e) => {
		setCommentContent(e.target.value);
	};

	//Show/unshow replies
	const [collapsedReplies, setCollapsedReplies] = useState(true);
	const toggleColapsed = () => {
		setCollapsedReplies(!collapsedReplies);
	};

	const edit_button = <BsPencil />;
	const delete_button = <BsXSquare />;

	return (
		<Aux>
			<CommentSUI>
				<CommentSUI.Avatar as="a" src={comment.author_image} />
				<CommentSUI.Content>
					<CommentSUI.Author
						as="a"
						onClick={() => history.push(`/profile/${comment.author}`)}
					>
						{comment.author_name}
					</CommentSUI.Author>
					<CommentSUI.Metadata>
						<span>{moment(comment.created_date).fromNow()}</span>
						{onEdit ? (
							<CommentSUI.Actions>
								<CommentSUI.Action
									className="delete-btn"
									onClick={() => {
										setShowDeleteModal(true);
										setDeleteCommentId(comment.id);
									}}
								>
									{delete_button}
								</CommentSUI.Action>
							</CommentSUI.Actions>
						) : null}
					</CommentSUI.Metadata>
					{!onEdit ? (
						<Aux>
							<CommentSUI.Text>{comment.comment_content}</CommentSUI.Text>
							<CommentSUI.Actions>
								<CommentSUI.Action onClick={toggleColapsed}>
									<CommentSUI.Metadata>
										{comment.reply_count}{' '}
										{comment.reply_count > 1 ? 'Replies' : 'Reply'}{' '}
										{collapsedReplies ? null : <FiChevronDown />}
									</CommentSUI.Metadata>
								</CommentSUI.Action>
								{userId === comment.author ? (
									<CommentSUI.Action
										className="edit-btn"
										onClick={toggleEditor}
									>
										{edit_button}
									</CommentSUI.Action>
								) : null}
							</CommentSUI.Actions>
						</Aux>
					) : (
						<Form reply onSubmit={() => update(comment.id, commentContent)}>
							<Form.TextArea
								style={{ height: 60, marginTop: -10 }}
								onChange={onChangeComment}
								value={commentContent}
							/>
							<BSButton
								type="submit"
								size="sm"
								className="mr-2 update-comment-button"
							>
								Update
							</BSButton>
							<BSButton onClick={toggleEditor} variant="secondary" size="sm">
								Cancel
							</BSButton>
						</Form>
					)}
				</CommentSUI.Content>
				<CommentSUI.Group collapsed={collapsedReplies}>
					<Replies commentId={comment.id} loadReplies={!collapsedReplies} />
				</CommentSUI.Group>
			</CommentSUI>
		</Aux>
	);
};

export default Comment;
