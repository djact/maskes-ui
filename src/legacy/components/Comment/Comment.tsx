import React, { useState } from 'react'
import { useHistory } from 'react-router-dom'
import { Comment as CommentSUI, Form } from 'semantic-ui-react'
import { Button as BSButton } from 'react-bootstrap'
import { FiChevronDown } from 'react-icons/fi'
import { BsPencil, BsXSquare } from 'react-icons/bs'
import moment from 'moment'
import Aux from '../../hoc/Aux/Aux'
import './Comment.css'
import Replies from '../../containers/Replies/Replies'

interface ISingleComment {
    id: string
    author: string
    author_name: string
    author_image: string
    comment_content: string
    reply_count: number
    created_date: string
}

interface IComment {
    comment: ISingleComment
    update: any
    userId: string
    setDeleteCommentId: any
    setShowDeleteModal: any
}

const Comment = (props: IComment) => {
    const { comment, update, userId, setDeleteCommentId, setShowDeleteModal } =
        props

    const history = useHistory()

    //Edit Comment Form
    const [onEdit, setOnEdit] = useState(false)
    const toggleEditor = () => {
        setOnEdit(!onEdit)
    }

    const [commentContent, setCommentContent] = useState(
        comment && comment.comment_content ? comment.comment_content : ''
    )
    const onChangeComment = (e) => {
        setCommentContent(e.target.value)
    }

    //Show/unshow replies
    const [collapsedReplies, setCollapsedReplies] = useState(true)
    const toggleColapsed = () => {
        setCollapsedReplies(!collapsedReplies)
    }

    const edit_button = <BsPencil />
    const delete_button = <BsXSquare />

    return (
        comment && (
            <Aux>
                <CommentSUI>
                    <CommentSUI.Avatar as="a" src={comment.author_image} />
                    <CommentSUI.Content>
                        <CommentSUI.Author
                            as="a"
                            onClick={() =>
                                history.push(`/profile/${comment.author}`)
                            }
                        >
                            {comment.author_name}
                        </CommentSUI.Author>
                        <CommentSUI.Metadata>
                            <span>
                                {moment(comment.created_date).fromNow()}
                            </span>
                            {onEdit ? (
                                <CommentSUI.Actions>
                                    <CommentSUI.Action
                                        className="delete-btn"
                                        onClick={() => {
                                            setShowDeleteModal(true)
                                            setDeleteCommentId(comment.id)
                                        }}
                                    >
                                        {delete_button}
                                    </CommentSUI.Action>
                                </CommentSUI.Actions>
                            ) : null}
                        </CommentSUI.Metadata>
                        {!onEdit ? (
                            <Aux>
                                <CommentSUI.Text>
                                    {comment.comment_content}
                                </CommentSUI.Text>
                                <CommentSUI.Actions>
                                    <CommentSUI.Action onClick={toggleColapsed}>
                                        <CommentSUI.Metadata>
                                            {comment.reply_count}{' '}
                                            {comment.reply_count > 1
                                                ? 'Replies'
                                                : 'Reply'}{' '}
                                            {collapsedReplies ? null : (
                                                <FiChevronDown />
                                            )}
                                        </CommentSUI.Metadata>
                                    </CommentSUI.Action>
                                    {userId === comment.author ? (
                                        <CommentSUI.Action
                                            className="edit-btn"
                                            onClick={toggleEditor}
                                        >
                                            {edit_button}
                                        </CommentSUI.Action>
                                    ) : null}
                                </CommentSUI.Actions>
                            </Aux>
                        ) : (
                            <Form
                                reply
                                onSubmit={() =>
                                    update(comment.id, commentContent)
                                }
                            >
                                <Form.TextArea
                                    style={{ height: 60, marginTop: -10 }}
                                    onChange={onChangeComment}
                                    value={commentContent}
                                />
                                <BSButton
                                    type="submit"
                                    size="sm"
                                    className="mr-2 update-comment-button"
                                >
                                    Update
                                </BSButton>
                                <BSButton
                                    onClick={toggleEditor}
                                    variant="secondary"
                                    size="sm"
                                >
                                    Cancel
                                </BSButton>
                            </Form>
                        )}
                    </CommentSUI.Content>
                    <CommentSUI.Group collapsed={collapsedReplies}>
                        <Replies
                            commentId={comment.id}
                            loadReplies={!collapsedReplies}
                        />
                    </CommentSUI.Group>
                </CommentSUI>
            </Aux>
        )
    )
}

export default Comment
