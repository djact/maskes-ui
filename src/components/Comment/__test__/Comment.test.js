import React from 'react';
import { shallow, mount } from 'enzyme';
import Comment from '../Comment';

describe('Comment', () => {
	it('should render comment correctly without props', () => {
		const component = shallow(<Comment />);
		expect(component).toMatchSnapshot();
	});

	it('should render comment correctly with passed in props', () => {
		const comment = {
			id: '1',
			author: '1',
			author_name: 'Duc',
			author_image: 'images/test.jpg',
			comment_content: 'test comment',
			reply_count: 10,
			created_date: new Date().toISOString()
		};
		const commentData = {
			comment: comment,
			update: () => {},
			userId: '1',
			setDeleteCommentId: () => {},
			setShowDeleteModal: () => {}
		};

		const component = shallow(<Comment {...commentData} />);

		expect(component).toMatchSnapshot();
	});
});
