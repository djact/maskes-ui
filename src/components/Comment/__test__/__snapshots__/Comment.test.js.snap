// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment should render comment correctly with passed in props 1`] = `
<aux>
  <Comment>
    <CommentAvatar
      as="a"
      src="images/test.jpg"
    />
    <CommentContent>
      <CommentAuthor
        as="a"
        onClick={[Function]}
      >
        Duc
      </CommentAuthor>
      <CommentMetadata>
        <span>
          a few seconds ago
        </span>
      </CommentMetadata>
      <aux>
        <CommentText>
          test comment
        </CommentText>
        <CommentActions>
          <CommentAction
            as="a"
            onClick={[Function]}
          >
            <CommentMetadata>
              10
               
              Replies
               
            </CommentMetadata>
          </CommentAction>
          <CommentAction
            as="a"
            className="edit-btn"
            onClick={[Function]}
          >
            <BsPencil />
          </CommentAction>
        </CommentActions>
      </aux>
    </CommentContent>
    <CommentGroup
      collapsed={true}
    >
      <Connect(Replies)
        commentId="1"
        loadReplies={false}
      />
    </CommentGroup>
  </Comment>
</aux>
`;

exports[`Comment should render comment correctly without props 1`] = `""`;
